test/unit/test_configcommon.py

```diff
@@ -64,3 +64,63 @@ def test_is_current_real_user_owner(autosubmit_config: Callable, owner):
     else:
         as_conf.experiment_data["AS_ENV_CURRENT_USER"] = "dummy"
     assert as_conf.is_current_real_user_owner == owner
+
+
+def test_clean_dynamic_variables_special_variables(autosubmit_config: Callable) -> None:
+    """Test cleaning special variables.
+
+    In Autosubmit configuration parser, variables that follow the pattern ``^%...%`` are
+    special variables, e.g. ``^%EXPID%``.
+
+    This test verifies that ``as_conf.special_variables`` is "cleaned", in the sense that
+    when the ``clean_dynamic_variables`` is called with the right argument, the special
+    variables dictionary contains only the special dynamic variables - nothing more.
+    """
+    in_the_end = True
+
+    as_conf: AutosubmitConfig = autosubmit_config(expid='a000', experiment_data={})
+    dynamic_variables, pattern, start_long = as_conf._initialize_variables(in_the_end=in_the_end)
+
+    as_conf.special_dynamic_variables = {
+        'popeye_eats': 'spinach',
+        'penguin_eats': 'fish',
+        'thor_eats': '%^DYNAMIC_1%',
+        'floki_eats': '%^DYNAMIC_2%',
+        'jaspion_eats': '%PLACEHOLDER%'
+    }
+
+    as_conf.clean_dynamic_variables(pattern, in_the_end=in_the_end)
+
+    # All other variables, that are not special, have been removed! So we should
+    # have only thor_eats and floki_eats here.
+    assert len(as_conf.special_dynamic_variables) == 2
+    assert 'thor_eats' in as_conf.special_dynamic_variables
+    assert 'floki_eats' in as_conf.special_dynamic_variables
+
+
+def test_clean_dynamic_variables(autosubmit_config: Callable) -> None:
```

Comment:
Initially I had a single test, with parameters. But if you look at how the function `clean_dynamic_variables` is designed, you should see that it's a bit odd to test. You pass an argument that defines whether you are cleaning dynamic or special variables.

If the variable is true, then you are cleaning special, and the `self.special_variables` must be used for testing; otherwise, the `self.dynamic_variables` is used. I started passing the name of the variable as a parameters... and then it started to get a bit convoluted for what's supposed to be a simple test, so I thought it'd be easier to split the test for now.

But in the future I believe we can rethink this function to make it simpler, e.g. note too that we call a function of `as_conf` to give us the `pattern`... and then later we give the `pattern` back to `as_conf`. That `pattern` is never used anywhere else, so it can exist within `as_conf`, without users having to retrieve it to give it back. A future improvement.